### PR TITLE
Added option whether to show terminal reuse alert

### LINKED ISSUE #42993: Allow user to disable "Terminal will be reused by tasks, press any key to close it" message.
10b14c6
- VSCode Version: 1.20 Insider 3f8cf5311b631c5bc395df1e8b9c795358c1ef80
- OS Version: macOS 10.13.4 beta

Steps to Reproduce:

1. Run a build task that outputs to the integrated terminal. VSCode should display "Terminal will be reused by tasks, press any key to close it" message at the end of the build output.

This message takes up three lines of console output (newlines before and after). When working on a laptop screen with the terminal pane at the bottom, it is proving to be quite annoying. I understand that the message is educational, but having learned how it works I never need to see it again, and I want the real estate back. Moreover, the message is visually distracting when I am looking at long lists of build errors.

Thanks!

### PR DESCRIPTION
**Pull request by oriash93**
Allow user to disable "Terminal will be reused by tasks, press any key to close it" message.

fixes #42993

### CODE REVIEW

File: src/vs/workbench/parts/terminal/electron-browser/terminal.contribution.ts [hunk 1]
```diff
@@ -193,6 +193,11 @@ configurationRegistry.registerConfiguration({
 			'type': 'boolean',
 			'default': false
 		},
+		'terminal.integrated.hideTerminalReuseAlert': {
```
Comment (Tyriar) [inline src/vs/workbench/parts/terminal/electron-browser/terminal.contribution.ts]: The `terminal.integrated` namespace currently doesn't have anything related to tasks/debug in it AFAIK. @dbaeumer is there a tasks settings namespace?

Comment (dbaeumer) [reply src/vs/workbench/parts/terminal/electron-browser/terminal.contribution.ts]: Yes, everything in a tasks.json is basically a setting. So whether the message is shown or not should be a task settings and a global setting in the tasks.json. 

The corresponding schema for this is here: https://github.com/Microsoft/vscode/blob/master/src\vs\workbench\parts\tasks\electron-browser\jsonSchema_v2.ts#L7

**Comment by oriash93** [comment]
I hope I got it right this time.

**Comment by dbaeumer** [comment]
I looked at the PR and it goes into the right direction. Since tasks have already a presentation property I opt to include it there and not make it a first level attribute. Something:

```
"presentation": {
    ....
    "reuseMessage": boolean
}
```

The default should be true although not a falsify value. But it makes more sense then negating the key.

And you only need to adopt the V2 schema since the presentation attribute is only supported on V2.

**Comment by oriash93** [comment]
@dbaeumer, I force-pushed the whole branch, but take a look at the last commit.

**Comment by dbaeumer** [comment]
Code looks OK to me. Can you add a test to ensure the configuration is passed correctly.

**Comment by dbaeumer** [comment]
@oriash93 I need to push this to Mai. I was extremely busy with other things in April. Sorry for that.

**Comment by dbaeumer** [comment]
Sorry for moving it again. I will promise to look into it first thing in June.

**Comment by dbaeumer** [comment]
I merged in the PR however I will tweak it a little. Mostly rename `reuseMessage` to `showReuseMessage`.

**Comment by gwk** [comment]
Is this available in the latest insiders build? I can't seem to get it to work but maybe I'm missing something.

**Comment by dbaeumer** [comment]
I should be in today's insider.

**Comment by buggymcbugfix** [comment]
Has this not landed in the current release version yet? I'm getting "Property message is not allowed" for `"reuseMessage"`.

**Comment by dbaeumer** [comment]
The property name is `"showReuseMessage"`

**Comment by umrashrf** [comment]
```
{
    // See https://go.microsoft.com/fwlink/?LinkId=733558
    // for the documentation about the tasks.json format
    "version": "2.0.0",
    "tasks": [
        {
            "label": "clean",
            "type": "shell",
            "windows": {
                "options": {
                    "shell": {
                        "executable": "cmd.exe",
                        "args": [
                            "/d", "/c"
                        ]
                    }
                },
                "command": "del /S /Q *.pyc *.log"
            },
            "presentation": {
                "showReuseMessage": false,
            }
        }
    ]
}
```

**Comment by rustyx** [comment]
@dbaeumer so this just hides the message "Press any key to close the terminal.", but user still has to press a key to reveal their terminal??

**Comment by dbaeumer** [comment]
@rustyx you can still set presentation.reveal to 'never' which will not bring the terminal to the front.

**Pull request merged**